applicatie/main/routes.py

```diff
@@ -19,37 +20,82 @@ def index():
 
 @bp.route("/matrix", methods=['GET', 'POST'])
 def matrix(jsonbestand):
-    """Laad de pagina met een overzicht."""
+    """Laad de pagina met een overzicht van de data.
+    Voer eerst diverse controles uit
+    en geef evt. foutmeldingen terug:
+        - json data bestand inlezen
+        - json schema ophalen van repo
+        - json data controleren aan json schema
+        - json definitie bestand ophalen van repo
+        - TODO: json data controleren aan het definitiebestand
+    """
     if jsonbestand:
-        # json bestand inlezen
-        databestand, foutmeldingen_databestand = ophalen_databestand(jsonbestand)
+        # json data bestand inlezen
+        databestand, fouten_databestand = ophalen_databestand(jsonbestand)
+        if fouten_databestand:
+            return render_template("index.html", errormessages=fouten_databestand)
 
-        if foutmeldingen_databestand:
-            return render_template("index.html", errormessages=foutmeldingen_databestand)
+        # json schema ophalen van repo
+        bestandsnaam = 'iv3_data_schema_v1_0.json'
+        schemabestand, fouten_schemabestand = ophalen_bestand_van_repo(REPO_PATH, bestandsnaam, 'schemabestand')
+        if fouten_schemabestand:
+            return render_template("index.html", errormessages=fouten_schemabestand)
 
-        # definitiebestand ophalen
-        metadata = databestand['metadata']
-        definitiebestand, foutmeldingen_definitiebestand = ophalen_definitiebestand(metadata)
+        # json data controleren aan json schema
+        fouten_schemacontrole = controle_met_schema(databestand, schemabestand)
+        if fouten_schemacontrole:
+            return render_template("index.html", errormessages=fouten_schemacontrole)
 
-        if foutmeldingen_definitiebestand:
-            return render_template("index.html", errormessages=foutmeldingen_definitiebestand)
+        # json definitie bestand ophalen van repo
+        metadata = databestand['metadata']
+        overheidslaag = metadata['overheidslaag']
+        boekjaar = metadata['boekjaar']
+        bestandsnaam = 'iv3_definities_' + overheidslaag + '_' + boekjaar + '.json'
+        definitiebestand, fouten_definitiebestand = ophalen_bestand_van_repo(REPO_PATH, bestandsnaam, 'definitiebetand')
+        if fouten_definitiebestand:
+            return render_template("index.html", errormessages=fouten_definitiebestand)
 
+        # json data controleren aan het definitiebestand
         sjabloon_meta = definitiebestand['metadata']
+        # TODO: nog niet klaar
+
+        # alle controles zijn uitgevoerd.
+        # nu data aggregeren en tonen op het scherm
+        try:
```

Comment:
Deze try/except is wat algemeen. Bij het controleren van het schema test je als het goed is of je databestand over data en lasten beschikt. Daarna moet je bestand goed zijn en is try/except hier overbodig.

Reply:
aangepast